javascript/javascript2/week2/script.js

```diff
@@ -0,0 +1,124 @@
+document.getElementById("years").innerHTML = filteredYears;
+
+
+//4.Create a new array that has an extra key called tag. The tag is based on the rating: Good (>= 7), Average (>= 4 and < 7), Bad (< 4)
+
+const taggedMovies = movies.map((movie) => {
+  if (movie.rating >= 7) {
+    movie.tag = "Good";
+  }
+  else if (movie.rating >= 4 && movie.rating < 7){
+    movie.tag = "Average";
+  }
+  else {
+    movie.tag = "Bad";
+  }
+});
+console.log(taggedMovies);
+document.getElementById("newTags").innerHTML = `
+  <ol>
+    ${taggedMovies}
+  </ol>
+`;
+
+//5.Using chaining, first filter the movies array to only contain the movies rated higher than 6.
+//Now map the movies array to only the rating of the movies.
+
+let ratedMovies = movies.filter((movie) => movie.rating > 6).map(movie => movie.rating);
+console.log(ratedMovies);
+document.getElementById("ratedSix").innerHTML = `
+  <ol>
+    ${ratedMovies}
+  </ol>
+`;
+
+//6. Count the total number of movies containing any of following keywords: Surfer, Alien or Benjamin. So if there were 3 movies that contained Surfer, 1 with Alien and 2 with Benjamin, you would return 6.
+//Can you make sure the search is case insensitive?
+
+const totalMovies = movies.filter((movie) => {
+  let title = movie["title"].toLowerCase()
+  if (title.includes("surfer") || title.includes("alien") || title.includes("benjamin")) {
+  return true
+  }
+}).length;
+console.log(totalMovies);
+
+//7. Create an array of movies where a word in the title is duplicated.
+//Fx "Star Wars: The Clone Wars" the word Wars is duplicated.
+function duplicated(array) {
+  const titleWithDuplicated = array.filter((movie) => {
+    if (new Set(movie.title.toLowerCase().split(' ')).size !== (movie.title.toLowerCase().split(' ')).length) {
+      return movie;
+    }
+  });
+console.log(titleWithDuplicated);
+}
+duplicated(movies);
+document.getElementById("duplicatedMovies").innerHTML = `
+  <ol>
+    ${titleWithDuplicated}
+  </ol>
+`;
+
+//8. Calculate the average rating of all the movies using reduce. Optional
+
+let totalMovieRating = movies.reduce((accumulator, movie) => accumulator + movie.rating, 0);
+let averageRating = totalMovieRating / movies.length;
+let toFixedRating = averageRating.toFixed(1);
+console.log(averageRating.toFixed(1));
+document.getElementById("averageRatingMovies").innerHTML = `
+  <ol>
+    ${toFixedRating}
+  </ol>
+`;
+
+//9. Count the total number of Good, Average and Bad movies using reduce. A return could fx be {goodMovies: 33, averageMovies: 45, goodMovies: 123} Optional
+
+
+let tags = movies.reduce((accumulator, movie) => {
```

Comment:
that is a nive way of using object withing reduce! 
The only think is this is not going to work becasue the `tag` property on the movies dos not exist in the initial array, first you would need to create a new array where movies has those tags and then run your reduce or change the condition. You could have three separate funcions defined in your script for determining if a movie is good or bad and use them here and in the task where you make an array with tags.